client/my-sites/plans-features-main/hooks/use-is-free-plan-custom-domain-upsell-enabled.ts

```diff
@@ -0,0 +1,23 @@
+import { useMemo } from '@wordpress/element';
```

Comment:
This is just a file movement. And a renaming of the function/file ( from client/my-sites/plans-features-main/components/free-plan-paid-domain-dialog.tsx ) to reflect the fact that the universal state of the  `FreePlanCustomDomainUpsell` will be calculated here.